sam.c

```diff
@@ -179,6 +226,14 @@ sam_hdr_t *sam_hdr_dup(const sam_hdr_t *h0)
         h->text[h->l_text] = '\0';
     }
 
+    if (h0->sdict) {
+        // We have a reason to use new API, so build it so we
```

Comment:
@jkbonfield As discussed, this would be better addressed inside `sam_hdr_update_target_arrays`.